cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -641,17 +640,14 @@ impl<T: Config> Pallet<T> {
 		});
 	}
 
-	fn enqueue_xcmp_messages(
+	fn enqueue_xcmp_messages<'a>(
 		sender: ParaId,
-		xcms: &[BoundedVec<u8, MaxXcmpMessageLenOf<T>>],
+		xcms: &[BoundedSlice<'a, u8, MaxXcmpMessageLenOf<T>>],
 		meter: &mut WeightMeter,
 	) -> Result<(), ()> {
 		let QueueConfigData { drop_threshold, .. } = <QueueConfig<T>>::get();
-		let batches_footprints = T::XcmpQueue::get_batches_footprints(
-			sender,
-			xcms.iter().map(|xcm| xcm.as_bounded_slice()),
-			drop_threshold,
-		);
+		let batches_footprints =
+			T::XcmpQueue::get_batches_footprints(sender, xcms.iter().copied(), drop_threshold);
```

Comment:
Weird that `BoundedSlice` does not implement `Copy`, give that it is just a reference.